### PR TITLE
enable https requests with ssl verification

### LINKED ISSUE #27581: DruidOperator only allows to make HTTP spec submissions
### Apache Airflow Provider(s)

apache-druid

### Versions of Apache Airflow Providers

_No response_

### Apache Airflow version

2.3.3

### Operating System

Amazon Linux 2

### Deployment

Virtualenv installation

### Deployment details

_No response_

### What happened

We have been using Imply Druid for aggregating data and only HTTPS requests are allowed to submit specs in our druid cluster.
Considering using DruidOperator for doing this job, we faced a problem that the operator only has support for HTTP requests. Even if the HTTP connection is set with `https` schema, inside the definition of the operator it doesn't consider at all.

### What you think should happen instead

I think that what should be done is that when `schema` is set in the HTTP connection used by the DruidOperator, it should consider on what type of request is going to be done.

### How to reproduce

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by viniolivieri**
This Pull Request has the objective to enable HTTPS requests via DruidOperator using the HTTPConnection attribute `schema`. Also it adds the feature of enabling HTTPS requests with SSL verification.

closes: [27581](https://github.com/apache/airflow/issues/27581)

It is my first contribution of all time in an open source project and all feedbacks are welcome!

Thanks,
Vini
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/apache/druid/hooks/druid.py [hunk 1]
```diff
@@ -48,13 +48,15 @@ def __init__(
         druid_ingest_conn_id: str = "druid_ingest_default",
         timeout: int = 1,
         max_ingestion_time: int | None = None,
+        use_ssl: bool = False,
```
Comment (o-nikolas) [inline airflow/providers/apache/druid/hooks/druid.py:51]: Can you update the doc string above to include this new param?

File: airflow/providers/apache/druid/hooks/druid.py [hunk 2]
```diff
@@ -82,12 +84,29 @@ def get_auth(self) -> requests.auth.HTTPBasicAuth | None:
         else:
             return None
 
+    def get_ssl_path(self) -> str | None:
+        """Get ssl path defined in extras from druid connection"""
+        conn = self.get_connection(self.druid_ingest_conn_id)
+        ssl_path = conn.extra_dejson.get("ssl_path", None)
```
Comment (o-nikolas) [inline airflow/providers/apache/druid/hooks/druid.py:90]: Is `None` an acceptable value to the `verify` arg on `requests.post(...)`? If not, perhaps we should raise an exception at this point if the user is trying to use ssl, but no path was provided.

File: airflow/providers/apache/druid/hooks/druid.py [hunk 3]
```diff
@@ -82,12 +84,29 @@ def get_auth(self) -> requests.auth.HTTPBasicAuth | None:
...
+        return ssl_path
+
     def submit_indexing_job(self, json_index_spec: dict[str, Any] | str) -> None:
         """Submit Druid ingestion job"""
         url = self.get_conn_url()
 
         self.log.info("Druid ingestion spec: %s", json_index_spec)
-        req_index = requests.post(url, data=json_index_spec, headers=self.header, auth=self.get_auth())
+
+        """Enable SSL Certification if needed."""
```
Comment (o-nikolas) [inline airflow/providers/apache/druid/hooks/druid.py:99]:
```suggestion
        # Enable SSL certification if requested.
```

File: tests/providers/apache/druid/hooks/test_druid.py [hunk 4]
```diff
@@ -148,13 +148,25 @@ def test_submit_timeout(self, m):
...
         get_conn_value.extra_dejson = {"endpoint": "ingest"}
         mock_get_connection.return_value = get_conn_value
         hook = DruidHook(timeout=1, max_ingestion_time=5)
         assert hook.get_conn_url() == "https://test_host:1/ingest"
 
+    @patch("airflow.providers.apache.druid.hooks.druid.DruidHook.get_ssl_path")
+    def test_get_ssl_path(self, mock_get_connection):
+        get_conn_value = MagicMock()
+        get_conn_value.host = "test_host"
+        get_conn_value.conn_type = "https"
```
Comment (o-nikolas) [inline tests/providers/apache/druid/hooks/test_druid.py:163]: didn't you add the `schema` field for this?

**Comment by o-nikolas** [comment]
@viniolivieri just checking in, are you still working on this PR?

**Comment by viniolivieri** [comment]
Hey @o-nikolas yes! I am just in a little bit hurry in another thing, but this week I will update it with the suggestion + fix tests.
Thanks.

**Comment by o-nikolas** [comment]
> Hey @o-nikolas yes! I am just in a little bit hurry in another thing, but this week I will update it with the suggestion + fix tests. Thanks.

Hey @viniolivieri, any update on this one?

**Comment by o-nikolas** [comment]
Closing this PR as stale, we can re-open if the original author has time to get back to this one.

**Pull request closed without merging**